CNV-22038: VM definition links in VME manifest (#59420)
openshift/openshift-docs: modules/virt-retrieving-vm-definition-from-vme-manifest.adoc

. Review the `status.links` stanza, which is divided into `external` and `internal` sections. Note the `manifests.url` fields within each section:
+
.Example output

[source,yaml]
----
apiVersion: export.kubevirt.io/v1alpha1
kind: VirtualMachineExport
metadata:
  name: example-export
spec:
  source:
    apiGroup: "kubevirt.io"
    kind: VirtualMachine
    name: example-vm
  tokenSecretRef: example-token
status:
...
  links:
    external:
...
      manifests:
      - type: all
        url: https://vmexport-proxy.test.net/api/export.kubevirt.io/v1alpha1/namespaces/example/virtualmachineexports/example-export/external/manifests/all <1>
      - type: auth-header-secret
        url: https://vmexport-proxy.test.net/api/export.kubevirt.io/v1alpha1/namespaces/example/virtualmachineexports/example-export/external/manifests/secret <2>
    internal:
...
      manifests:
      - type: all
        url: https://virt-export-export-pvc.default.svc/internal/manifests/all <3>
      - type: auth-header-secret
        url: https://virt-export-export-pvc.default.svc/internal/manifests/secret
  phase: Ready
  serviceName: virt-export-example-export
----

> <1> Access the `VirtualMachine` manifest, `DataVolume` objects, if present, and a config map that contains the public certificate for the external URL's ingress or route.

Reviewer: ```suggestion
<1> Access the `VirtualMachine` manifest, `DataVolume` manifest, if present, and a `ConfigMap` that contains the public certificate for the external URL's ingress or route.
```

<2> Access a secret containing a header that is compatible with Containerized Data Importer (CDI). The header contains a text version of the export token.

> <3> Access the `VirtualMachine` manifest, `DataVolume` objects, if present, and a config map that contains the certificate authority (CA) of the internal URL's export server.

Reviewer: ```suggestion
<3> Access the `VirtualMachine` manifest, `DataVolume` manifest, if present, and a `ConfigMap` that contains the certificate for the internal URL's export server.
```

Author: Changed "objects" to "manifest" in both locations, but I changed "config map" to "`ConfigMap` manifest" to follow our API object formatting guidelines ([link if you're curious](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting)). Let me know if this works.

> 

Reviewer: I'm just wondering if we should add the instructions on how to get the manifests from the manifests links, because it requires having the certificates in the target cluster.
I'm using these steps:

Source cluster: save the certificates. 
Target cluster: copy cacert.crt to the target cluster
`oc get vmexport example-export -o jsonpath={.status.links.external.cert} > cacert.crt
`

Source cluster: decode the token and save it:
`oc get secret export-token-example-export -o jsonpath={.data.token} | base64 --decode > token_decode
`

Target cluster: get secret headers yaml using a secret manifest and a `token_decode` from the source cluster. Create the Secret

`curl --cacert cacert.crt https://virt-exportproxy-openshift-cnv.test.net/api/export.kubevirt.io/v1alpha1/namespaces/default/virtualmachineexports/example-export/external/manifests/secret -H "x-kubevirt-export-token:<token_decode>" -H "Accept:application/yaml"
`

Author: This sounds really important to include; thank you! I'm a little confused about how I should sequence this, though. Am I understanding this correctly? Thanks so much @jpeimer 

1. Copy cacert.crt to the target cluster (by running the command you provided from the target cluster, IIUC)
2. Decode the token and save it (by running the command you provided from the source cluster)
3. Existing step: get the custom resource
4. Existing step: examine the CR and identify the manifest URLs
5. From the target cluster, create the Secret by running the `curl --cacert ...` command, referencing a Secret manifest URL (that you got from the YAML in step 4) and the token_decode (that you created in step 2)
6. Access the manifest URLs to your heart's content and/or apply them to the cluster

.Procedure

. To access the manifests, you must first copy the certificates from the source cluster to the target cluster.

.. Log in to the source cluster.

.. Save the certificates to the file `cacert.crt` by running the following command:
+
[source,terminal]
----
$ oc get vmexport <export_name> -o jsonpath={.status.links.external.cert} > cacert.crt <1>
----
<1> Replace `<export_name>` with the `metadata.name` value from the `VirtualMachineExport` object.

.. Copy the `cacert.crt` file to the target cluster.

. Decode the token in the source cluster and save it to the file `token_decode` by running the following command:
+
[source,terminal]
----
$ oc get secret export-token-<export_name> -o jsonpath={.data.token} | base64 --decode > token_decode <1>
----

> <1> Replace `<export_name>` with the `metadata.n

Reviewer: <1> Replace `<export_name>` with the `metadata.name` value from the `VirtualMachineExport` object.
(maybe we can even remove it and '<1>' from the line above because we say it in line 32, but whatever you think is better)

Author: I have no idea how that got cut off! 😹 Great catch. I think I'll fix it and leave it in, just because it's a different command and folks might not be reading super closely
